### PR TITLE
fix: Add missing triggers to Map#Card axioms in Boogie prelude

### LINKED ISSUE #2380: Incompleteness in map cardinality equality
The following without uncommenting the commented line fails to verify:
```dafny
method M(t: map<int, int>, k: int)
  requires k in t
{
  var lt := |t|;
  var lt' := |t - {k}|;
  // var _ := |(t - {k}).Keys|;
  assert lt == lt' + 1;
}
```

Uncommenting the line makes it verify, because it triggers an axiom that is needed.

### PR DESCRIPTION
**Pull request by cpitclaudel**
Closes #2380.

@atomb , can you show me how to run a performance regression run on this?  I'm not sure how it will impact our customers.

### CODE REVIEW

File: Test/git-issues/git-issue-2380.dfy [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method M(t: map<int, int>, k: int)
+  requires k in t
+{
+  assert |t| == |t - {k}| + 1;
```
Comment (MikaelMayer) [inline Test/git-issues/git-issue-2380.dfy:7]: Could you assert something about the cardinality of the values and of the items? That way, you can also test your two other axioms

Comment (cpitclaudel) [reply Test/git-issues/git-issue-2380.dfy]: Good idea!  Done.

**Review by MikaelMayer — APPROVED** [review]
Fantastic ! Thanks for doing it.

**Comment by cpitclaudel** [comment]
Thanks for the approval!  Let's hold off on merging this PR until @prvshah51 helps me make sure that it doesn't cause performance regressions, though :)

**Comment by MikaelMayer** [comment]
I'm launching a regression test to see if we can merge this one.

**Comment by MikaelMayer** [comment]
Although these axioms are useful and sound, @atomb and I found that existing proofs would be slow down by up to 5%, so we don't know what to do at this point other than stall and not include these axioms.

**Review by atomb — APPROVED** [review]
After taking into account other recent changes, this now usually improves performance.

**Pull request merged**